### ISSUE TITLE
Syntax coloring inconsistent within a file

### ISSUE THREAD

**Issue by osialr** [intellij-rust/intellij-rust#1578]
I'm on Windows 10, IntelliJ IDEA 2017.2.1  Build #IC-172.3544.35 , Rust 1.19, and Rust plugin 0.1.0.2063.

I have a fairly colorful color scheme set up and it does not always get applied.  See for example this clip from rust-geo where the top and bottom impls are colored but the middle is not.

![Missing coloring](https://user-images.githubusercontent.com/16169732/28998575-c1c4ef0c-79fc-11e7-9629-53b3f3c000c6.png)

In another instance, the semantic highlighting colorization is applied, but further down the parameters are plain.

![Missing parameter rainbow](https://user-images.githubusercontent.com/16169732/28998597-4a849b26-79fd-11e7-9e4e-f9f7d4761bc3.png)

I'm not sure if this is related, but there appears to also be stack-overflow errors within the 
[log files](https://github.com/intellij-rust/intellij-rust/files/1202663/idea.txt)

**Comment by matklad** [comment]
Yeah, looks like the stackoverflow is the real culprint here, and it should be fixed in the new version which was released just moments ago! Please reopen if the issue persists after the update :) 

Thanks for the report!

**Comment by osialr** [comment]
@matklad Thanks. I updated the plugin, but the same stackoverflow is occurring the latest plugin and in master (0d4be39f4)  I don't have permission to reopen this issue, could you open it?

This is the file I'm looking at
https://github.com/georust/rust-geo/blob/master/src/types.rs


I put a println for `element`, `boundSub`, `trait`, and `subst` before the loop at https://github.com/intellij-rust/intellij-rust/blob/4f39ddd99b4fe53623473040dc0bf1a27cb44713/src/main/kotlin/org/rust/lang/core/types/ty/TyTypeParameter.kt#L35

And the same lines keep printing over and over:

```
element=RsTraitItemImpl(TRAIT_ITEM)

boundSubt={<Self as Num>::FromStrRadixErr=<Self as Num>::FromStrRadixErr, <Self as Add>::Output=<Self as Add>::Output, <Self as Mul>::Output=<Self as Mul>::Output, <Self as Sub>::Output=<Self as Sub>::Output, <Self as Div>::Output=<Self as Div>::Output, Self=Self, <Self as Neg>::Output=<Self as Neg>::Output}, 

trait=BoundElement(element=RsTraitItemImpl(TRAIT_ITEM), subst={<Self as Num>::FromStrRadixErr=<T as Num>::FromStrRadixErr, <Self as Add>::Output=<T as Add>::Output, <Self as Mul>::Output=<T as Mul>::Output, <Self as Sub>::Output=<T as Sub>::Output, <Self as Div>::Output=<T as Div>::Output, Self=T, <Self as Neg>::Output=<T as Neg>::Output})

subst={<Self as Num>::FromStrRadixErr=<T as Num>::FromStrRadixErr, <Self as Add>::Output=<T as Add>::Output, <Self as Mul>::Output=<T as Mul>::Output, <Self as Sub>::Output=<T as Sub>::Output, <Self as Div>::Output=<T as Div>::Output, Self=T, <Self as Neg>::Output=<T as Neg>::Output}
```

**Comment by matklad** [comment]
@vlad20012 take a look please: looks like anoter unify with SO

**Comment by vlad20012** [comment]
The issue should be fixed by #1592. Please, check it on the latest [nightly](https://intellij-rust.github.io/install.html) Intellij-rust build (or build it from master).

**Comment by osialr** [comment]
That fixed it. Thanks!
